Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -244,8 +244,10 @@ public NodejsProjectNode(NodejsProjectPackage package)
         }
 
         protected override void FinishProjectCreation(string sourceFolder, string destFolder) {
+            bool foundTypeScriptFile = false;
             foreach (MSBuild.ProjectItem item in this.BuildProject.Items) {
                 if (String.Equals(Path.GetExtension(item.EvaluatedInclude), NodejsConstants.TypeScriptExtension, StringComparison.OrdinalIgnoreCase)) {
+                    foundTypeScriptFile = true;
```

Comment:
That sounds good. I think your point about `node_modules` and `bower_components` exposes a flaw with the existing logic. Updating both of the conditionals to handle those cases seems like the correct behavior, as right now, we can end up enabling typescript for projects that only include TS files in those locations.

Will update the review shortly to fix this and also use `EnableTypeScript` by dispatching to the UI thread.